05-JWT-Basics/preferred/public/browser-app.js

```diff
@@ -0,0 +1,58 @@
+const formDOM = document.querySelector('form');
+const usernameInputDOM = document.querySelector('.username-input');
+const passwordInputDOM = document.querySelector('.password-input');
+const formAlertDOM = document.querySelector('.form-alert');
+const resultDOM = document.querySelector('.result');
+const btnDOM = document.querySelector('#data');
+const tokenDOM = document.querySelector('.token');
+
+formDOM.addEventListener('submit', async (e) => {
+  formAlertDOM.classList.remove('text-success');
+  tokenDOM.classList.remove('text-success');
+
+  e.preventDefault();
+  const name = usernameInputDOM.value;
+  const password = passwordInputDOM.value;
+
+  try {
+    const { data } = await axios.post('/api/v1/logon', { name, password });
+
+    formAlertDOM.style.display = 'block';
+    formAlertDOM.textContent = data.message;
```

Comment:
In your `preferred` app's logon route/controller, you're only returning the `token` and not returning any message. So `data.message` will always be undefined here. 

Not a big deal since this is just some text that displays "you're logged in successfully" or something like that. But in that case we can just remove this line and line 23

Reply:
@akosasante thank you !